src/bin.tsx

```diff
@@ -16,36 +19,61 @@ program
   .storeOptionsAsProperties(false)
   .passCommandToAction(false)
   .description('Deletes old GitLab pipelines')
-  .arguments('<gitlab-url>')
-  .arguments('<project-id>')
-  .arguments('<access-token>')
+  .arguments('[gitlab-url]')
+  .arguments('[project-id]')
+  .arguments('[access-token]')
   .option('-d --days <days>', 'older than days', '30')
   .option('--trace', 'show stack traces for errors when possible', false)
-  .action((gitlabUrl: string, projectId: string, accessToken: string, options: Record<string, unknown>) => {
-    const parsedProjectId = parseInt(projectId, 10);
-    const days = typeof options.days === 'string' ? parseInt(options.days, 10) : 30;
-    const startDate = new Date();
-    const listPipelinesFunction = listPipelines({ getRequest, gitlabUrl, projectId: parsedProjectId, accessToken });
-    const deletePipelineFunction = deletePipeline({
-      deleteRequest,
-      gitlabUrl,
-      projectId: parsedProjectId,
-      accessToken,
-    });
-    render(
-      <Main
-        gitlabUrl={gitlabUrl}
-        projectId={parsedProjectId}
-        accessToken={accessToken}
-        days={days}
-        startDate={startDate}
-        exit={exit}
-        listPipelines={listPipelinesFunction}
-        filterPipelinesByDate={filterPipelinesByDate}
-        deletePipeline={deletePipelineFunction}
-        showStackTraces={options.trace === true}
-      />,
-    );
-  });
+  .action(
+    async (
+      gitlabUrlArgument: string | undefined,
+      projectIdArgument: string | undefined,
+      accessTokenArgument: string | undefined,
+      options: Record<string, unknown>,
+    ) => {
+      const cliArguments: CliArguments = {
+        gitlabUrl: gitlabUrlArgument,
+        projectId: projectIdArgument ? parseInt(projectIdArgument, 10) : undefined,
+        accessToken: accessTokenArgument,
+        days: typeof options.days === 'string' ? parseInt(options.days, 10) : 30,
+        trace: options.trace === true,
+      };
+      const explorer = cosmiconfig('gitlab-pipeline-deleter');
+      const config = await loadConfig('./glpd.config.js', explorer);
```

Comment:
Does it automatically load the file relative to the CWD?

Reply:
Good question 🤔 Would it make more sense to have a `path.resolve(__dirname, 'glpd.config.js')` here?

Reply:
No, `__dirname` doesn’t make sense, since you want to load the file relative to the user CWD folder and not the `gitlab-pipeline-delete` folder within `node_modules`.

Reply:
Ah sorry I meant `process.cwd()`. So `path.resolve(process.cwd(), 'glpd.config.js')` ?